internal/pkg/manifest/errors.go

```diff
@@ -98,6 +98,37 @@ func (e *errFieldMutualExclusive) Error() string {
 	return fmt.Sprintf(`must specify one, not both, of "%s" and "%s"`, e.firstField, e.secondField)
 }
 
+type errGracePeriodSpecifiedMoreThanOnce struct {
+	firstField  string
+	secondField string
+	index       int
+}
+
+func (e *errGracePeriodSpecifiedMoreThanOnce) Error() string {
+	if strings.Contains(e.firstField, "additional_rules") {
+		return fmt.Sprintf(`cannot define "grace_period" in "http.additional_rules[%d].healthcheck.grace_period"`, e.index)
+	} else if strings.Contains(e.firstField, "additional_listeners") {
```

Comment:
This is very confusing because it has nothing to do with the name of  `errGracePeriodSpecifiedMoreThanOnce`. I feel like we need a different error for this case. can we use `errFieldMutualExclusive` if they specify `grace_period` in both places and a new error type if they specify `grace_period` in additional rules?

Looking forward the key is really about "decoupling" (so that we can make changes in one place; easier to maintain; and less error-prone): we should avoid mixing things together just because they belong to the same "topic" but focusing on what they do (functionalities). For example, in this errors file `errContainersExposingSamePort` is a probably ok because it is specific but we have to do it this way (the error message is specifically about port). Whereas `errSpecifiedBothIngressFields` is a bad one because the error msg has nothing to do with ingress fields and we should've reused `errFieldMutualExclusive` (even if we need new recommendation we could make it based on `errFieldMutualExclusive`)

Reply:
yeah that sounds good to me. I have changed error message when customer define `grace_period` in both ALB and NLB to errFieldMutualExclusive. And renamed existing `errGracePeriodSpecifiedMoreThanOnce` to `errGracePeriodSpecifiedInAdditionalField` when customer define `grace_period` in additional rules or listeners. Do let me know how does that look to you?